APIs/getDailyWeather.js

```diff
@@ -0,0 +1,58 @@
+const express = require("express");
+const router = express.Router();
+const axios = require("axios");
+
+const { getDailyWeatherResponse } = require("../DTOs/index");
+const { APPID , FORECAST_URL, weekday } = require("../common/index");
+
+const getCurrentHour = () => {
+  const date = new Date();
+  const hours = date.getHours();
+  const remainder = hours % 3;
+  const currentHour = remainder === 0 ? hours : remainder === 1 ? hours - 1 : hours + 1;
```

Comment:
the logic looks complicated can't we just use 
`const currentHour = 3 - remainder`?

Reply:
we have to get the nearest hour, this does not gives the nearest hour.

Reply:
make date variable parametrized

Reply:
`/**
* explain why we are doing
*/

let currentHour = hours;

const CurrentHourIsNotDivisibleByThree = remainder !== 0;

if(CurrentHourIsNotDivisibleByThree)
{
  const isCurrentHourNearestToNextHour = remainder > 1;
  if(isCurrentHourNearestToNextHour) currentHour += 1;
  else currentHour -= 1;
}
`

Reply:
```
/**
* explain why we are doing
*/

let currentHour = hours;

const CurrentHourIsNotDivisibleByThree = remainder !== 0;

if(CurrentHourIsNotDivisibleByThree)
{
  const isCurrentHourNearestToNextHour = remainder > 1;
  if(isCurrentHourNearestToNextHour) currentHour += 1;
  else currentHour -= 1;
}

```